src/test/scala/com/fulcrumgenomics/commons/collection/ParIteratorTest.scala

```diff
@@ -0,0 +1,47 @@
+package com.fulcrumgenomics.commons.collection
+
+import com.fulcrumgenomics.commons.util.UnitSpec
+
+class ParIteratorTest extends UnitSpec {
+  "ParIterator" should "map the same as a non-parallel map" in {
+    val xs  = Range.inclusive(1, 100)
+    val par = ParIterator(xs.iterator, chunkSize=8, threads=4, chunkBuffer=Some(3))
+    val out = par.map(_ + 1).map(_ * 2).toAsync().toIndexedSeq
+    val exp = xs.map(_ + 1).map(_ * 2)
+    out shouldBe exp
```

Comment:
How do you feel about this verbosity?
```suggestion
    out contain theSameElementsInOrderAs exp
```